src/dagmc.cpp

```diff
@@ -331,24 +277,364 @@ void load_dagmc_geometry()
     }
 
     // add to global array and map
-    model::surfaces.emplace_back(s);
-    model::surface_map[s->id_] = i;
+
+    auto in_map = model::surface_map.find(s->id_);
+    if (in_map == model::surface_map.end()) {
+      model::surface_map[s->id_] = model::surfaces.size();
+    } else {
+      warning(fmt::format("DAGMC Surface IDs: {}", dagmc_ids_for_dim(2)));
+      fatal_error(fmt::format("Surface ID {} exists in both Universe {} "
+                              "and the CSG geometry.", s->id_, this->id_));
+    }
+
+    model::surfaces.emplace_back(std::move(s));
+  } // end surface loop
+}
+
+std::string
+DAGUniverse::dagmc_ids_for_dim(int dim) const
+{
+  // generate a vector of ids
+  std::vector<int> id_vec;
+  int n_cells = dagmc_instance_->num_entities(dim);
+  for (int i = 1; i <= n_cells; i++) {
+    id_vec.push_back(dagmc_instance_->id_by_index(dim, i));
+  }
+
+  // sort the vector of ids
+  std::sort(id_vec.begin(), id_vec.end());
+
+  // generate a string representation of the ID range(s)
+  std::stringstream out;
+
+  int i = 0;
+  int start_id = id_vec[0];
+  int stop_id;
+  while (i < n_cells) {
+    stop_id = id_vec[i];
+
```

Comment:
quite comment light, i presume you've making a list like 1-100, 101, 105-109?

Reply:
Correct. I'll add some more explanation here.